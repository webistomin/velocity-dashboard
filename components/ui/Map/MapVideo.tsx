import { VueComponent } from 'types/vue-components';
import { Component } from 'nuxt-property-decorator';
import { VNode } from 'vue';

@Component({
  name: 'MapVideo',
})
export default class MapVideo extends VueComponent {
  render(): VNode {
    return (
      <div class='map__video'>
        <button class='map__video-play-btn btn'>
          <img class='map__video-play-image image' src='/img/interior.jpg' alt='Video' />
          <svg-icon name='icon-play' width={18} height={20} />
        </button>
        <div class='map__video-content'>
          <div class='map__video-info'>
            <strong class='map__video-name'>Activate interior video</strong>
            <span class='map__video-place'>37-27 74th Street</span>
          </div>
          <img class='map__video-camera image' src='/img/camera.svg' width={24} height={24} alt='Camera' />
        </div>
      </div>
    );
  }
}
