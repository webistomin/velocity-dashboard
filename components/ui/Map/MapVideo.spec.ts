import { shallowMount } from '@vue/test-utils';
import MapVideo from './MapVideo';

describe('MapVideo', () => {
  const factory = (propsData = {}, options = {}) => {
    return shallowMount(MapVideo, {
      propsData: {
        ...propsData,
      },
      stubs: ['svg-icon'],
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });
});
