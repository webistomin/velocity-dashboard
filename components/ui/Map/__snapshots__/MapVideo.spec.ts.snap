// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapVideo Matches snapshot 1`] = `
"<div class=\\"map__video\\"><button type=\\"button\\" class=\\"map__video-play-btn btn\\"><img src=\\"/img/interior.jpg\\" alt=\\"Video\\" class=\\"map__video-play-image image\\">
    <svg-icon-stub name=\\"icon-play\\" width=\\"18\\" height=\\"20\\"></svg-icon-stub>
  </button>
  <div class=\\"map__video-content\\">
    <div class=\\"map__video-info\\"><strong class=\\"map__video-name\\">Activate interior video</strong><span class=\\"map__video-place\\">37-27 74th Street</span></div><img src=\\"/img/camera.svg\\" width=\\"24\\" height=\\"24\\" alt=\\"Camera\\" class=\\"map__video-camera image\\">
  </div>
</div>"
`;
